tests/ops/api/v1/endpoints/test_messaging_endpoints.py

```diff
@@ -0,0 +1,761 @@
+        return (V1_URL_PREFIX + MESSAGING_SECRETS).format(
+            config_key=messaging_config.key
+        )
+
+    @pytest.fixture(scope="function")
+    def payload(self):
+        return {
+            MessagingServiceSecrets.MAILGUN_API_KEY.value: "1345234524",
+        }
+
+    def test_put_config_secrets_unauthenticated(
+        self, api_client: TestClient, payload, url
+    ):
+        response = api_client.put(url, headers={}, json=payload)
+        assert 401 == response.status_code
+
+    def test_put_config_secrets_wrong_scope(
+        self, api_client: TestClient, payload, url, generate_auth_header
+    ):
+        auth_header = generate_auth_header([MESSAGING_READ])
+        response = api_client.put(url, headers=auth_header, json=payload)
+        assert 403 == response.status_code
+
+    def test_put_config_secret_invalid_config(
+        self, api_client: TestClient, payload, generate_auth_header
+    ):
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+        url = (V1_URL_PREFIX + MESSAGING_SECRETS).format(config_key="invalid_key")
+        response = api_client.put(url, headers=auth_header, json=payload)
+        assert 404 == response.status_code
+
+    def test_update_with_invalid_secrets_key(
+        self, api_client: TestClient, generate_auth_header, url
+    ):
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+        response = api_client.put(url, headers=auth_header, json={"bad_key": "12345"})
+
+        assert response.status_code == 400
+        assert response.json() == {
+            "detail": [
+                "field required ('mailgun_api_key',)",
+                "extra fields not permitted ('bad_key',)",
+            ]
+        }
+
+    def test_put_config_secrets(
+        self,
+        db: Session,
+        api_client: TestClient,
+        payload,
+        url,
+        generate_auth_header,
+        messaging_config,
+    ):
+        auth_header = generate_auth_header([MESSAGING_CREATE_OR_UPDATE])
+        response = api_client.put(url, headers=auth_header, json=payload)
+        assert 200 == response.status_code
+
+        db.refresh(messaging_config)
+
+        assert json.loads(response.text) == {
+            "msg": "Secrets updated for MessagingConfig with key: my_mailgun_messaging_config.",
+            "test_status": None,
+            "failure_reason": None,
+        }
+        assert (
+            messaging_config.secrets[MessagingServiceSecrets.MAILGUN_API_KEY.value]
+            == "1345234524"
+        )
+
+
+class TestPutMessagingConfigSecretTwilioEmail:
+    @pytest.fixture(scope="function")
+    def url(self, messaging_config_twilio_email) -> str:
+        return (V1_URL_PREFIX + MESSAGING_SECRETS).format(
+            config_key=messaging_config_twilio_email.key
+        )
+
+    @pytest.fixture(scope="function")
+    def payload(self):
+        return {MessagingServiceSecrets.TWILIO_API_KEY.value: "23p48btcpy14b"}
```

Comment:
Just confirming this and the other keys here are fake keys

Reply:
yup, they are fake